API/projectClass/tableUtilities.m

```diff
@@ -44,21 +44,46 @@ function addRow(obj, varargin)
 
         function removeRow(obj, row)
             % Removes a row from the table. The expected input is an
-            % integer or integer array.
+            % integer/string or integer/string array.
             % NOTE that an input such as [1 3] leads to multiple rows
             % being removed from the table
             %
             % varTable.removeRow(2);
             arguments
                 obj
-                row {mustBePositive, mustBeInteger}
+                row
             end
 
-            if row > obj.rowCount
-                throw(exceptions.indexOutOfRange(sprintf('Row index %d out of range 1 - %d', row, obj.rowCount)));
+            % Arrange parameters into a cell array
+            if isa(row, 'double')
+
+                indices = sort(row, 'descend');
+
+            elseif isText(row) || iscell(row)
+
+                row = cellstr(row);
+                indices = zeros(1, length(row));
+
+                rowNames = obj.getNames;
+                for i = 1:length(row)
+                    currentRow = row{i};
+                    indices(i) = obj.findRowIndex(currentRow, rowNames, 'Unrecognised parameter name');
+                end
+
+            else
+
+                throw(exceptions.invalidType('Unrecognised Row'))
+
             end
-            
-            obj.varTable(row, :) = [];
+
+            indices = sort(indices, 'descend');
```

Comment:
Since the indices are sorted here, the sort on line 60 is not needed. Also the for loop below is not needed since indices is sorted.

Reply:
If the input is a cell array of multiple row names, the resulting indices array will not be sorted, so it is needed here. There's also no guarantee that a list of rows provided is valid so I think the for loop is still needed too.

Reply:
Ahh, sorry. Misread here. I'll take out the sort on line 60 too.